utils/errors/errors.go

```diff
@@ -0,0 +1,114 @@
+package errors
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/gogo/protobuf/grpc"
+	"github.com/tendermint/tendermint/libs/log"
+	"golang.org/x/xerrors"
+	grpc2 "google.golang.org/grpc"
+)
+
+type causer interface {
+	Cause() error
+}
+
+// Is checks if the error is a (wrapped) error of the given type
+func Is[T error](err error) bool {
+	switch e := err.(type) {
+	case nil:
+		return false
+	case T:
+		return true
+	case causer:
+		return Is[T](e.Cause())
+	case xerrors.Wrapper:
+		return Is[T](e.Unwrap())
+	default:
+		return false
+	}
+}
```

Comment:
```suggestion
func Is[T error](err error) bool {
	for {
		switch e := err.(type) {
		case nil:
			return false
		case T:
			return true
		case causer:
			err = e.Cause()
		case xerrors.Wrapper:
			err = e.Unwrap()
		default:
			return false
		}
	}
```
Doesn't look too bad?